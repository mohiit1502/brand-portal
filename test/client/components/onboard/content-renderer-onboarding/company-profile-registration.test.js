// /* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
// import "core-js/stable";
// import "regenerator-runtime/runtime";
// import React from "react";
// import {Provider} from "react-redux";
// import {BrowserRouter} from "react-router-dom";
// import renderer from "react-test-renderer";
// import Cookies from "electrode-cookies";
// import {mockFetch, testStore, update} from "../../../../../src/client/utility/TestingUtils";
// import {mount} from "enzyme";
// import CompanyProfileRegistration from "../../../../../src/client/components/onboard/content-renderer-onboarding/company-profile-registration";
// import profile from "../../../mocks/userProfile";
// import MODALSMETA from "../../../../../src/client/config/contentDescriptors/modals-meta";
// import FORMFIELDMETA from "../../../../../src/client/config/formsConfig/form-field-meta";
// import applicationDetails from "../../../mocks/applicationDetails";
//
// let store;
// jest.mock("electrode-cookies");
//
// const setUpRender = () => {
//   const mockStore = {
//     content: {
//       metadata: {...MODALSMETA, ...FORMFIELDMETA},
//     },
//     company: {onboardingDetails: applicationDetails},
//     user: {profile}
//   };
//   store = testStore(mockStore);
//   return renderer.create(<Provider store={store}><CompanyProfileRegistration /></Provider>);
// };
//
// const setUpMount = (isEditContext) => {
//   profile.context = isEditContext ? "edit" : "new";
//   const mockStore = {
//     content: {
//       metadata: {...MODALSMETA, ...FORMFIELDMETA},
//     },
//     company: {onboardingDetails: applicationDetails},
//     user: {profile}
//   };
//   store = testStore(mockStore);
//   return mount(<Provider store={store}><BrowserRouter><CompanyProfileRegistration /></BrowserRouter></Provider>);
// };
//
// describe("CompanyProfileRegistration test container", () => {
//   let wrapper;
//   beforeEach(() => {
//     jest.restoreAllMocks();
//     jest.resetAllMocks();
//     Cookies.get.mockImplementation(jest.fn());
//   });
//
//   it("should render the CompanyProfileRegistration successfully", () => {
//     wrapper = setUpRender();
//     expect(wrapper.toJSON()).toMatchSnapshot();
//   });
//
//   it("should trigger input change events and actions for edit context", async () => {
//     jest.useFakeTimers();
//     const companyName = "test-company";
//     const clientType = "seller";
//     mockFetch({
//       data: {
//         shouldMap: true,
//         [`/api/company/availability?companyName=${companyName}&clientType=${clientType}`]: {unique: true, name: companyName}
//       }
//     });
//     wrapper = setUpMount(true);
//     await update(wrapper);
//     wrapper.find(".form-control-companyName").at(0).simulate('change', {target: {value: companyName}});
//     // wrapper.find(".dropdown-item").at(0).simulate('click');
//     // wrapper.find("div.btn.btn-sm.btn-block.btn-primary").at(0).simulate('click');
//     // wrapper.find("button.btn.btn-sm.btn-block.cancel-btn.text-primary").at(0).simulate('click');
//     // wrapper.find(".form-control-comments").at(0).simulate('change', {target: {value: "comment-12345432-comment"}});
//     jest.runAllTimers();
//     jest.useRealTimers();
//   });
//
//   it("should trigger input change events and actions for new context", async () => {
//     jest.useFakeTimers();
//     const companyName = "test-company";
//     const clientType = "seller";
//     mockFetch({
//       data: {
//         shouldMap: true,
//         [`/api/company/availability?companyName=${companyName}&clientType=${clientType}`]: {unique: true, name: companyName}
//       }
//     });
//     wrapper = setUpMount(false);
//     await update(wrapper);
//     wrapper.find(".form-control-companyName").at(0).simulate('change', {target: {value: companyName}});
//     // wrapper.find("div.btn.btn-sm.btn-block.btn-primary").at(0).simulate('click');
//     // wrapper.find("button.btn.btn-sm.btn-block.cancel-btn.text-primary").at(0).simulate('click');
//     // wrapper.find(".form-control-comments").at(0).simulate('change', {target: {value: "comment-12345432-comment"}});
//     jest.runAllTimers();
//     jest.useRealTimers();
//   });
// });
