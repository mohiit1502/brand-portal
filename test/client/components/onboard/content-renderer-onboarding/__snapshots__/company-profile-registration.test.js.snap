// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyProfileRegistration test container should render the CompanyProfileRegistration successfully 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-lg-10 col-md-8 col-12 pl-5 pr-0"
  >
    <div
      className="row title-row mb-4 pl-4"
    >
      <div
        className="col"
      >
        <div
          className="company-registration-title"
        >
          Company Information
        </div>
        <div
          className="company-registration-subtitle mt-2"
        >
          Complete the form based on the information in the trademark registration.
        </div>
      </div>
    </div>
    <form
      className="company-reg-form mb-4 pl-4"
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className=" col"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-0
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-companyName custom-input-element"
              disabled={false}
              id="companyreg-companyName-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={true}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="companyreg-companyName-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                Company Name
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              The company name should match the information in the trademark registration. 
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className=" col"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-0
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-address custom-input-element"
              disabled={false}
              id="companyreg-address-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={true}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="companyreg-address-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                Address
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              The company address should match the information in the trademark registration.
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-city custom-input-element"
              disabled={false}
              id="companyreg-city-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={true}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="companyreg-city-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                City
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
        <div
          className="col-6 col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text disabled mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-state custom-input-element"
              disabled={true}
              id="companyreg-state-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={true}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="companyreg-state-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                State
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-6 col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text disabled mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-zip custom-input-element"
              disabled={true}
              id="companyreg-zip-custom-input"
              maxLength={10}
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              pattern="^\\\\d{5}(?:[-]\\\\d{4})?$"
              required={true}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="companyreg-zip-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                ZIP
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
        <div
          className="col-6 col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text disabled mb-3
        "
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <input
              className="form-control form-control-country custom-input-element"
              disabled={true}
              id="companyreg-country-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={true}
              type="text"
              value="USA"
            />
            <label
              className="custom-input-label"
              htmlFor="companyreg-country-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                Country
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className=" col"
        >
          <div
            className="c-FileUploader form-row primary-file-upload mb-3"
          >
            <div
              className="col"
            >
              <div
                className="file-upload-title mb-2"
              >
                Optional: Attach additional documents if necessary (i.e. business registration)
                 
                <img
                  alt="i"
                  data-html="true"
                  data-placement="right"
                  data-toggle="tooltip"
                  src="question.svg"
                  title="<div data-reactroot=\\"\\"><ol class=\\"m-0 p-0\\"><ul class=\\"m-0 pl-3 text-left font-size-12\\"><li>Supported File Types: pdf, doc, docx, xls, xlsx, csv, jpeg, png</li><li>Maximum file size: 7 MB</li></ul></ol></div>"
                />
              </div>
              <label
                className="btn btn-sm btn-outline-primary upload-btn my-2 disabled"
              >
                Upload
                <input
                  accept="application/msword,.csv,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,application/vnd.ms-excel,application/vnd.openxmlformats-officedocument.wordprocessingml.document,text/plain,application/pdf,image/*"
                  className="d-none"
                  disabled={true}
                  onChange={[Function]}
                  onClick={[Function]}
                  type="file"
                />
              </label>
              
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="company-onboarding-button-panel text-right col"
        >
          <div
            className="c-ButtonsPanel form-row mt-3"
            data-test="button-panel"
          >
            <div
              className="col company-onboarding-button-panel text-right"
              data-test="button-div"
            >
              <button
                className="btn btn-sm btn-primary submit-btn px-3 ml-3"
                disabled={true}
                type="submit"
              >
                Continue
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
